specutils/spectra/xrayspectrum1d.py

```diff
@@ -0,0 +1,542 @@
+    def counts(self):
+        return self.flux
+
+    def assign_arf(self, arf_inp):
+        """
+        Assign an ARF object to the XraySpectrum1D object
+
+        Input
+        -----
+        arf_inp : string
+            File name for the ARF (FITS file)
+
+        Returns
+        -------
+        Modifies the XraySpectrum1D.arf attribute
+        """
+        if isinstance(arf_inp, str):
+            self.arf = ARF(arf_inp)
+        else:
+            self.arf = arf_inp
+        return
+
+    def assign_rmf(self, rmf_inp):
+        """
+        Assign an RMF object to the XraySpectrum1D object
+
+        Input
+        -----
+        rmf_inp : string
+            File name for the RMF (FITS file)
+
+        Returns
+        -------
+        Modifies the XraySpectrum1D.rmf attribute
+        """
+        if isinstance(rmf_inp, str):
+            self.rmf = RMF(rmf_inp)
+        else:
+            self.rmf = rmf_inp
+        return
+
+    def apply_resp(self, mflux, exposure=None):
+        """
+        Given a model flux spectrum, apply the response. In cases where the
+        spectrum has both an ARF and an RMF, apply both. Otherwise, apply
+        whatever response is in RMF.
+
+        The model flux spectrum *must* be created using the same units and
+        bins as in the ARF (where the ARF exists)!
+
+        Parameters
+        ----------
+        mflux : iterable
+            A list or array with the model flux values in ergs/keV/s/cm^-2
+
+        exposure : float, default None
+            By default, the exposure stored in the ARF will be used to compute
+            the total counts per bin over the effective observation time.
+            In cases where this might be incorrect (e.g. for simulated spectra
+            where the pha file might have a different exposure value than the
+            ARF), this keyword provides the functionality to override the
+            default behaviour and manually set the exposure time to use.
+
+        store_model_counts : bool
+            If True, the output will also be stored in self.model_counts
+
+        Returns
+        -------
+        model_counts : numpy.ndarray
+            The model spectrum in units of counts/bin
+
+        If no ARF file exists, it will return the model flux after applying the RMF
+        If no RMF file exists, it will return the model flux after applying the ARF (with a warning)
+        If no ARF and no RMF, it will return the model flux spectrum (with a warning)
+        """
+
+        if self.arf is not None:
+            mrate  = self.arf.apply_arf(mflux, exposure=exposure)
+        else:
+            #print("Caution: no ARF file specified")
+            mrate = mflux
+
+        if self.rmf is not None:
+            result = self.rmf.apply_rmf(mrate)
+        else:
+            #print("Caution: no RMF file specified")
+            result = mrate
+
+        return result
+
+## ----  Supporting response file objects
+
+class RMF(object):
```

Comment:
Please use the full name for the class

Reply:
I spelled out the class names to `AreaResponse` and `ResponseMatrix`.

However, these file types are so standard that I urge we keep most of the attribute and function names in the shortened version. X-ray astronomers even refer to them as "arfs" and "rmfs" verbally. I spelled out the acronyms in many of the docstrings to aid in clarity for a new user.